src/metric_definitions.go

```diff
@@ -5,7 +5,62 @@
 
 package main
 
-import "github.com/newrelic/infra-integrations-sdk/data/metric"
+import (
+	"reflect"
+
+	"github.com/newrelic/infra-integrations-sdk/data/metric"
+)
+
+// Definitions struct will contain the metrics that have to be collected.
+type Definitions struct {
+	Common              []Query `yaml:"common"`
+	Metrics             []Query `yaml:"metrics"`
+	ColumnFamilyMetrics []Query `yaml:"column_family_metrics"`
+}
+
+// FilterDefinitions filters the definitions of the metrics that have to be collected based on received config.
+func (d Definitions) FilterDefinitions(config Config) Definitions {
```

Comment:
The name seems to suggest that the definition is going to be filtered, but the receiver is not modified and we return a new Definition instance.
would be clearer to have something like "buildNewFilteredDefinition(definition, config)" (maybe now the name is even worse 😅 )

Reply:
What about `(d Definitions) FilteredCopy(config Config) Definitions` or something like that?